AutoCollection/Statsbeat.ts

```diff
@@ -134,39 +121,44 @@ class Statsbeat {
         this._instrumentation &= ~instrumentation;
     }
 
-    public countRequest(duration: number, success: boolean) {
+    public countRequest(category: number, endpoint: string, duration: number, success: boolean) {
         if (!this.isEnabled()) {
             return;
         }
-        this._totalRequestCount++;
-        this._intervalRequestExecutionTime += duration;
+        let counter: Network.NetworkStatsbeat = this._getNetworkStatsbeatCounter(category, endpoint);
+        counter.totalRequestCount++;
+        counter.intervalRequestExecutionTime += duration;
         if (success === false) {
-            this._totalFailedRequestCount++;
+            counter.totalFailedRequestCount++;
         }
         else {
-            this._totalSuccesfulRequestCount++;
+            counter.totalSuccesfulRequestCount++;
         }
+
     }
 
-    public countException() {
+    public countException(category: number, endpoint: string) {
```

Comment:
category and endpoint only matter when it's actually sending the exception count, right? I am not sure why incrementing the count care about these two input..

Reply:
Well but the counts are specific to a category/endpoint combination, Breeze could have 10 exceptions and Quickpulse 6, we don't want to mix counts in here, what are you proposing instead?

Reply:
it makes sense to me to track count at the category level.. it's crazy to count at the endpoint level.. there are lots of stamp specific endpoints for breeze.  we want to know exception count to breeze not at a specific stamp endpoint.. for live metric, it's the same host name.

Reply:
We don't expect customer to hit tons of endpoints, default would be something like this "dc.services.visualstudio.com", if redirect is triggered we will some other Statsbeat with endpoint "centralus-2.in.applicationinsights.azure.com", if customer provide custom endpointUrl during initialization we will get that as well, I'm just implementing what we discussed previously and I thought everyone agreed.

Reply:
As hector said, while breeze may have multiple servers (stamp) this endpoint is the one configured in the app not the stamp specific VM.
And I think that it does actually make sense to track exceptions (and all below) as the endpoint level as if we introduce a bug which causes an exception or the user is getting throttled etc this is all good information to help track down issues

Reply:
i think we might have a misunderstanding. i think we just want the total exception count.  for debugging, exception detail will be available in the exception table.  this count won't give much info about the actual cause of the exception. 

dc.services.visualstudio.com
centralus-2.in.applicationinsights.azure.com
are all breeze endpoint.. 

I am saying we should aggregate the exception count at breeze not at the host.  It doesn't offer much benefit except sending more Exception Count metrics than necessary.

Reply:
It won't be sending ```more Exception Count metrics``` as an application will generally only have 1 endpoint that it's sending too (excluding redirect scenario, at which point it just changes to the new one).

What it does change is the cardinality of the "complete" collection as it this is an additional dimension for the metrics.

Reply:
well, java agent supports live polling of the json config.. like azure sdk.. customers can change their connection string and sampling rate at runtime.. depends on how often the connection string gets updated.. this will trigger equivalent amount of the Count metrics and then multiplies by 6 (different network statsbeat metric count).  for regular customers who use single ikey throughout the lifetime of the app, category and endpoint won't get changed at all and there is no point to pass it in for each increment either.